### PR TITLE
fix: remove duplicate locations

### LINKED ISSUE #427: Remove duplicate locations in Search
# Requirements
Currently, multiple locations are being shown in the SearchBar menu. 

- [  ] Make it so the locations show once with a number of how many time they exist. Example would be `--- Yokohama (2) ---`

# Dependencies
None

### PR DESCRIPTION
**Pull request by NabbeunNabi**
Resolves #427 

## 🔧 What changed
I added a function that would make a lookup of location with key value pairs in order to make it so we can render based on location and how many doctors in that location. That is then used to render and search for doctors based on the value.

## 🧪 Testing instructions
None at the moment

## 📸 Screenshots

-   ### Before
image

-   ### After
image

### CODE REVIEW

File: components/SearchBar.vue [hunk 1]
```diff
@@ -2,92 +2,180 @@
...
 </template>
 
 <script setup lang="ts">
-import { ref, Ref, watchEffect } from 'vue'
-import { useSearchResultsStore } from '~/stores/searchResultsStore.js'
-import { useLocationsStore } from '~/stores/locationsStore.js'
-import { useSpecialtiesStore, SpecialtyDisplayOption } from '~/stores/specialtiesStore.js'
-import { Locale, Specialty } from "~/typedefs/gqlTypes.js"
-import { useLocaleStore, LocaleDisplay } from '~/stores/localeStore.js'
+import { ref, Ref, watchEffect } from "vue";
```
Comment (ermish) [inline components/SearchBar.vue:100]: We gotta fix this linting issue. We don't want to litter semicolons everywhere. 
I haven't seen it on other people's PRs yet, let's see if someone else's is also doing this or if it's just your machine. 🤔

File: components/SearchBar.vue [hunk 2]
```diff
@@ -2,92 +2,180 @@
...
+const languageDropdownOptions: Ref<LocaleDisplay[]> = ref(languageOptions);
 
-const selectedSpecialty: Ref<Specialty | String> = ref('')
-const selectedLocation: Ref<string> = ref('')
-const selectedLanguage: Ref<Locale | String> = ref('')
+const selectedSpecialty: Ref<Specialty | String> = ref("");
+const selectedLocation: Ref<string> = ref("");
+const selectedLanguage: Ref<Locale | String> = ref("");
+
+function createCityTable() {
```
Comment (ermish) [inline components/SearchBar.vue:138]: A table might be a bit of a stretch...maybe you could call it something like `createCityDisplayText` or `addCityCountToDisplayText` to make it more clear what's happening

File: components/SearchBar.vue [hunk 3]
```diff
@@ -2,92 +2,180 @@
...
+    let lookupCityCount: { [key: string]: number } = {};
+    const cities = locationsStore.citiesDisplayOptions;
+
+    for (let i = 0; i < cities.length; i++) {
```
Comment (ermish) [inline components/SearchBar.vue:142]: You can use a forEach loop instead  here to make it more readable

File: components/SearchBar.vue [hunk 4]
```diff
@@ -2,92 +2,180 @@
...
+        selectedSpecialty.value == "----Any----"
+            ? undefined
+            : (selectedSpecialty.value as Specialty);
+    const blankRemovedLanguage =
+        selectedLanguage.value == "----Any----"
+            ? undefined
+            : (selectedLanguage.value as Locale);
 
-    await searchResultsStore.search(blankRemovedLocation, blankRemovedSpecialty, blankRemovedLanguage)
+    await searchResultsStore.search(
```
Comment (ermish) [inline components/SearchBar.vue:175]: This formatting all looks completely different. Let's figure this out before we merge

**Review by ermish — CHANGES_REQUESTED** [review]
Solution looks great! Let's iron out a few details

**Comment by NabbeunNabi** [comment]
Ok! I will look and implement these. I will turn off my prettier when I work on this repo. I thought the linting would fix whatever my prettier was doing, but it just seems to be creating more problems than it is fixing. So I will just made sure that I am no throwing semicolons everywhere.

File: components/SearchBar.vue (hunk 4 above)
Comment (NabbeunNabi) [reply components/SearchBar.vue:175]: I put everything back the way it was

File: components/SearchBar.vue (hunk 3 above)
Comment (NabbeunNabi) [reply components/SearchBar.vue:142]: changed it to a` forEach` loop

File: components/SearchBar.vue (hunk 2 above)
Comment (NabbeunNabi) [reply components/SearchBar.vue:138]: changed the name of the function to your first suggestion `createCityDisplayText`

File: components/SearchBar.vue (hunk 1 above)
Comment (NabbeunNabi) [reply components/SearchBar.vue:100]: Turned off prettier and it seemed to not change and add all the semicolons! So hopefully I won't keep making this an issue. Sorry again

**Comment by NabbeunNabi** [comment]
I made a mistake rebasing so our original conversation commit isn't there but I made the changes requested.

File: components/SearchBar.vue [hunk 5]
```diff
@@ -79,6 +80,26 @@ const selectedSpecialty: Ref<Specialty | String> = ref('')
 const selectedLocation: Ref<string> = ref('')
 const selectedLanguage: Ref<Locale | String> = ref('')
 
+function createCityDisplayText() {
+    let cityDisplayTextObject: { [key: string]: number } = {}
+    const cities = locationsStore.citiesDisplayOptions
+
+    cities.forEach((city) => {
+        if (city === "----Any----") {
```
[the remark this answers is not in the document]
Comment (NabbeunNabi) [reply components/SearchBar.vue:88]: Oh! Great point! I'll update this tomorrow

**Review by theyokohamalife — APPROVED** [review]
LGTM 😎

**Comment by theyokohamalife** [comment]
@NabbeunNabi Looks good! I went ahead and approved it, but you should consider changing the PR title from "feat" to "fix" since it's fixing a 🐛 . Thank you!

**Comment by NabbeunNabi** [comment]
> @NabbeunNabi Looks good! I went ahead and approved it, but you should consider changing the PR title from "feat" to "fix" since it's fixing a 🐛 . Thank you!

Thank you and PR title updated

File: components/SearchBar.vue [hunk 6]
```diff
@@ -79,6 +80,28 @@ const selectedSpecialty: Ref<Specialty | String> = ref('')
 const selectedLocation: Ref<string> = ref('')
 const selectedLanguage: Ref<Locale | String> = ref('')
 
+const anyLocation = "----Any----"
```
Comment (ermish) [inline components/SearchBar.vue:83]: like the variable, but the name wasn't clear to me at first what it was.  (I read if the city is literally any location, then return)

```suggestion
const placeholderDisplayText = "----Any----"
```

File: components/SearchBar.vue [hunk 7]
```diff
@@ -79,6 +80,28 @@ const selectedSpecialty: Ref<Specialty | String> = ref('')
...
+
+function createCityDisplayText() {
+    let cityDisplayTextObject: { [key: string]: number } = {}
```
Comment (ermish) [inline components/SearchBar.vue:86]: while this is an efficient storage option, it's not very clear what values your using......you could use an array of objects here with two fields to make it a bit more clear

example

```js
const cityDisplayItems = [
  {
    cityText: 'dallas',
    cityCount: 2
  },
 {
    cityText: 'tokyo',
    cityCount: 5
  }
]
```

**Comment by ermish** [comment]
@NabbeunNabi your PR works and would be good enough to merge, but I added some extra thoughts to take it to the next level 🎚️

**Comment by NabbeunNabi** [comment]
I think the failed checks are something I do not know the key or have access to

image

**Review by ermish — APPROVED** [review]
beautiful ✨

**Pull request merged**